### ISSUE TITLE
Add API `Operator::spill(...)` ?

### ISSUE THREAD

**Issue by zhztheplayer** [facebookincubator/velox#4449]
### Description

Apache Spark controls spilling from its memory manager. It requires memory consumers support handle spill requests from Spark's memory manager. A typical spill invocation flow in Spark would like be:

operator acquires memory -> memory limit exceed -> request spill on all consumers -> memory space fits -> successful

**Comment by zhztheplayer** [comment]
@mbasmanova @xiaoxmeng 

Would you think this is possible for Velox? Or was there already some plan related?

**Comment by mbasmanova** [comment]
@zhztheplayer I wonder if it will be possible to implement this logic using memory arbitration: https://github.com/facebookincubator/velox/pull/4348

**Comment by xiaoxmeng** [comment]
> @mbasmanova @xiaoxmeng
> 
> Would you think this is possible for Velox? Or was there already some plan related?

@zhztheplayer  Yes, this function will be covered by the ongoing memory arbitrator work.

**Comment by zhztheplayer** [comment]
Thank you for working on memory arbitration! May I ask if this PR #4348 will contain all of the work or if it simply adds an arbitration API?
